libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Conway/ParametricSpec.hs

```diff
@@ -76,193 +82,142 @@ class
 
 instance IsConwayUniv fn => EraSpecTxOut MaryEra fn where
   irewardSpec = instantaneousRewardsSpec
   hasPtrs _proxy = lit True
-  correctTxOut = betterTxOutMary
+
+  -- correctTxOut = betterTxOutMary
   txOutValue_ x = sel @1 x
   txOutCoin_ x = maryValueCoin_ (sel @1 x)
+  txOutAddr_ x = sel @0 x
 
 instance IsConwayUniv fn => EraSpecTxOut AlonzoEra fn where
   irewardSpec = instantaneousRewardsSpec
   hasPtrs _proxy = lit True
-  correctTxOut = betterTxOutAlonzo
+
+  -- correctTxOut = betterTxOutAlonzo
   txOutValue_ x = sel @1 x
   txOutCoin_ x = maryValueCoin_ (sel @1 x)
+  txOutAddr_ x = sel @0 x
 
 instance IsConwayUniv fn => EraSpecTxOut BabbageEra fn where
   irewardSpec = instantaneousRewardsSpec
   hasPtrs _proxy = lit True
-  correctTxOut = betterTxOutBabbage
+
+  -- correctTxOut = betterTxOutBabbage
   txOutValue_ x = sel @1 x
   txOutCoin_ x = maryValueCoin_ (sel @1 x)
+  txOutAddr_ x = sel @0 x
 
 instance IsConwayUniv fn => EraSpecTxOut ConwayEra fn where
-  irewardSpec _ = constrained $ \ [var|irewards|] ->
+  irewardSpec _ _ = constrained $ \ [var|irewards|] ->
     match irewards $ \ [var|reserves|] [var|treasury|] [var|deltaRes|] [var|deltaTreas|] ->
       [ reserves ==. lit Map.empty
       , treasury ==. lit Map.empty
       , deltaRes ==. lit (DeltaCoin 0)
       , deltaTreas ==. lit (DeltaCoin 0)
       ]
   hasPtrs _proxy = lit False
-  correctTxOut = betterTxOutBabbage
+
+  -- correctTxOut = betterTxOutBabbage
   txOutValue_ x = sel @1 x
   txOutCoin_ x = maryValueCoin_ (sel @1 x)
+  txOutAddr_ x = sel @0 x
+
+-- ===========================================================================
+
+-- | An Era polymorhic Specification for type family TxOut
+txOutSpec ::
+  forall fn era.
+  EraSpecTxOut era fn =>
+  WitUniv era ->
+  Term fn (Map (Credential 'Staking) (KeyHash 'StakePool)) ->
+  Term fn (TxOut era) ->
+  Pred fn
+txOutSpec univ delegs txOut =
+  Block
+    [ assert $ 0 <. txOutCoin_ @era txOut
+    , assert $ txOutCoin_ @era txOut <=. fromIntegral (maxBound :: Word64)
+    , (caseOn (txOutAddr_ @era txOut))
+        -- Network -> Credential -> StakeRefernce -> Addr
+        ( branchW 2 $ \ [var|network|] [var|payCred|] [var|stakeref|] ->
+            [ witness univ payCred -- satisfies payCred (payCredSpec univ)
+            , assert $ network ==. lit Testnet
+            , satisfies stakeref (delegatedStakeReference delegs)
+            ]
+        )
+        -- BootstrapAddress -> Addr
+        (branchW 1 $ \bootstrapAddr -> satisfies bootstrapAddr (witBootstrapAddress univ))
+    ]
+
+-- | Generate random Stake references that have a high probability of being delegated.
+delegatedStakeReference ::
+  IsConwayUniv fn =>
+  Term fn (Map (Credential 'Staking) (KeyHash 'StakePool)) ->
+  Specification fn StakeReference
+delegatedStakeReference delegs =
+  constrained $ \ [var|ref|] ->
+    caseOn
+      ref
+      (branchW 9 $ \ [var|base|] -> member_ base (dom_ delegs))
+      (branchW 0 $ \_ptr -> False)
```

Comment:
Any reason why we disable delegation to a Ptr? I know that they are not resolved, but we do have them as an option, so I think it would be good to generate those occasionally as well.
```suggestion
      (branchW 1 $ \_ptr -> True)
```

Reply:
I am a little worried, that in the DState for Conway, there are no Ptrs, but probably won't affect things.